anitya/lib/versions/calver.py

```diff
@@ -53,6 +53,16 @@ def split_by_match(regex: str, string: str) -> Tuple[str, str]:
     return "", string
 
 
+class SplitResult(TypedDict, total=True):
```

Comment:
I'm not a fun of having multiple classes in one .py file, but this looks good. I didn't knew about the `TypedDict` till now.